src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -4106,6 +4106,10 @@ void StubGenerator::generate_compiler_stubs() {
 
   generate_chacha_stubs();
 
+  if ((UseAVX == 2) && EnableX86ECoreOpts && VM_Version::supports_avx2()) {
+    StubRoutines::_string_indexof = generate_string_indexof();
+  }
```

Comment:
What motivation for this extensive new code only for avx512? 30% is nice (for some cases) but it is enabled only for AVX2 and not for avx512 which all modern x86 CPUs have so the code will not be used.

Or it is typo?

Reply:
This is acceleration for AVX2, replacing the pcmpestri instruction which is microcoded on E-cores and causes significant performance impact.  I am working on a pared-down implementation and should update this PR in a couple of days.

Reply:
Thank you for explanation.